### PR TITLE
[BULK] Review configs

### PR DESCRIPTION
**Pull request by Deep1998**
Review config options for the bulk template.

### CODE REVIEW

**Comment by shanemeister** [comment]
Te

**Review by VardhanThigle — APPROVED** [review]

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java [hunk 1]
```diff
@@ -23,13 +23,11 @@ public interface SourceDbToSpannerOptions extends CommonTemplateOptions {
...
   @TemplateParameter.Text(
       order = 1,
-      optional = true,
       regexes = {"^.+$"},
       groupName = "Source",
       description = "Comma-separated Cloud Storage path(s) of the JDBC driver(s)",
       helpText = "The comma-separated list of driver JAR files.",
       example = "gs://your-bucket/driver_jar1.jar,gs://your-bucket/driver_jar2.jar")
-  @Default.String("")
   String getJdbcDriverJars();
```
Comment (VardhanThigle) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java:31]: Please retain the default.

The default "" implies, a user has compiled the jars into the build via say mvn imports, so no need to external class loading.

This is done for say UT, where we use derby dependency from mvn.

Comment (Deep1998) [reply v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java]: Done

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java [hunk 2]
```diff
@@ -71,6 +71,7 @@ public interface SourceDbToSpannerOptions extends CommonTemplateOptions {
   /* TODO: (support Sharding, PG namespaces) */
   @TemplateParameter.Text(
       order = 5,
+      regexes = {"(^[^\\/?%*:|\"<>.]{1,64}$)"},
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java:74]: This regex is too expensive for dataflow. 
`ERROR: (gcloud.dataflow.flex-template.run) FAILED_PRECONDITION: (406846f638a637e8): Template metadata regex '(^[^\/?%*:|"<>.]{1,64}$)' was too large. Max size is 1000 but was 1031.`

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/config/JdbcIOWrapperConfig.java [hunk 3]
```diff
@@ -134,12 +103,6 @@ public abstract static class Builder {
 
     public abstract Builder setJdbcDriverClassName(String value);
 
-    public abstract Builder setConnectionProperties(String value);
-
-    public abstract Builder setReconnectAttempts(Long value);
-
-    public abstract Builder setAutoReconnect(Boolean value);
```
Comment (bharadwaj-aditya) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/config/JdbcIOWrapperConfig.java:141]: update the javadocs to suggest to add these parameters in the jdbc url

**Review by bharadwaj-aditya — CHANGES_REQUESTED** [review]
There seem to be a few methods added here without tests. Can you add tests for those ?

**Pull request merged**